### PR TITLE
Fix Windows builds with OCaml >= 5.0

### LINKED ISSUE #6148: Cannot resolve symbols for win_create_process_native in opam-core on Windows
Detected in https://github.com/ocaml/opam-repository/pull/26345
```
 #=== ERROR while compiling opam-0install.0.4.4 ================================#
  # context     2.2.0 | win32/x86_64 | ocaml.5.2.0 | file://D:/a/opam-repository/opam-repository
  # path        D:\opamroot\default\.opam-switch\build\opam-0install.0.4.4
  # command     D:\opamroot\default\bin\dune.exe build -p opam-0install -j 3 @install
  # exit-code   1
  # env-file    D:\opamroot\log\opam-0install-6432-95eca8.env
  # output-file D:\opamroot\log\opam-0install-6432-95eca8.out
  ### output ###
  # File "bin/dune", line 2, characters 7-11:
  # 2 |  (name main)
  #            ^^^^
  # (cd _build/default && D:\opamroot\default\bin\ocamlopt.opt.exe -w -40 -g -o bin/main.exe D:\opamroot\default\lib\re\re.cmxa D:\opamroot\default\lib\stdlib-shims\stdlib_shims.cmxa D:\opamroot\default\lib\ocamlgraph\graph.cmxa D:/opamroot/default/lib/ocaml\unix\unix.cmxa D:\opamroot\default\lib\opam-core\stubs/opam_stubs_win32.cmxa -I D:\opamroot\default\lib\opam-core\stubs D:\opamroot\default\[...]
  # ** Cannot resolve symbols for D:\opamroot\default\lib\opam-core\opam_core.a(opamStubs.o):
  #  win_create_process_native
  # ** Cannot resolve symbols for D:\opamroot\default\lib\opam-core\opam_core.a(opamProcess.o):
  #  win_create_process_native
  # File "caml_startup", line 1:
  # Error: Error during linking (exit code 2)
```

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Fixes https://github.com/ocaml/opam/issues/6148

### CODE REVIEW

File: .github/workflows/ci.ml [hunk 1]
```diff
@@ -15,7 +15,10 @@ open Lib
 
 let ocamls = [
   (* Fully supported versions *)
-   "4.08.1"; "4.09.1"; "4.10.2"; "4.11.2"; "4.12.1"; "4.13.1"; "5.0.0"; "5.1.1"; "4.14.1";
+   "4.08.1"; "4.09.1"; "4.10.2"; "4.11.2"; "4.12.1"; "4.13.1";
+   "5.0.0"; "5.1.1";
+   "4.14.1"; "5.2.0"; (* The last elements of the list after 4.14 be used as
```
Comment (kit-ty-kate) [inline .github/workflows/ci.ml:20]: oops i forgot MSVC is only supported by 5.3 not 5.2. It might be worth waiting for 5.3.0~alpha1

**Comment by kit-ty-kate** [comment]
OCaml 5.3 support requires a new release of extlib first actually (see https://github.com/ygrek/ocaml-extlib/pull/73)

**Comment by kit-ty-kate** [comment]
I split the Github action part in https://github.com/ocaml/opam/pull/6192 to be merged later

File: src/core/dune [hunk 2]
```diff
@@ -19,6 +19,14 @@
 (rule
   (copy opamStubsTypes.ml opamStubsTypes.mli))
 
+(rule
+ (enabled_if (< %{ocaml_version} "5.0"))
+ (action (copy opamStubs.ocaml4.ml opamStubs.win32.ml)))
+
+(rule
+ (enabled_if (>= %{ocaml_version} "5.0"))
+ (action (copy opamStubs.ocaml5.ml opamStubs.win32.ml)))
```
Comment (dra27) [inline src/core/dune:28]:
```suggestion
 (action (copy# opamStubs.ocaml4.ml opamStubs.win32.ml)))

(rule
 (enabled_if (>= %{ocaml_version} "5.0"))
 (action (copy# opamStubs.ocaml5.ml opamStubs.win32.ml)))
```

**Review by dra27 — APPROVED** [review]
`copy#` means that in the unlikely event of a syntax error, Dune continues to report the correct source file. I have a suggestion in https://github.com/dra27/opam/commit/0608ec5cb4e467479e0e9084b7ee3b631919cb8f which eliminates the code duplication between the two OCaml files, but I'm not sure it's necessarily better, given that the file changes very rarely.

**Pull request merged**